### PR TITLE
rgw/rgw_swift_auth.cc: using string::back() instead as the C++11 recommend

### PR DESCRIPTION
**Pull request by liuyuhong66**
Signed-off-by: liuyuhong <liuyuhong@cmss.chinamobile.com>

### CODE REVIEW

**Comment by liuyuhong66** [comment]
@Jing-Scott @rzarzynski please help me to have a look, tks

**Comment by Jing-Scott** [comment]
looks good!

File: src/rgw/rgw_swift_auth.cc [hunk 1]
```diff
@@ -304,7 +304,7 @@ ExternalTokenEngine::authenticate(const std::string& token,
   }
 
   std::string auth_url = g_conf->rgw_swift_auth_url;
-  if (auth_url[auth_url.length() - 1] != '/') {
+  if (auth_url.back() != '/') {
```
Comment (rzarzynski) [inline src/rgw/rgw_swift_auth.cc]: May we have a comment saying that`ExternalTokenEngine::is_applicable` guarantees `rgw_swift_auth_url` is non-empty?

Anyway, it's only a suggestion. Even now the commit is perfectly mergeable.

Comment (yehudasa) [inline src/rgw/rgw_swift_auth.cc]: @liuyuhong66  just modify it to `if (!auth_url.empty() && auth_url.back() != '/')`

**Comment by liuyuhong66** [comment]
@rzarzynski  @yehudasa    i did some changes，tks

**Comment by rzarzynski** [comment]
LGTM! It will be merged soon.

File: src/rgw/rgw_swift_auth.cc [hunk 2]
```diff
@@ -304,7 +304,7 @@ ExternalTokenEngine::authenticate(const std::string& token,
   }
 
   std::string auth_url = g_conf->rgw_swift_auth_url;
-  if (auth_url[auth_url.length() - 1] != '/') {
+  if (!auth_url.empty() && auth_url.back() != '/') {
```
Comment (tchaikov) [inline src/rgw/rgw_swift_auth.cc]: i don't think the check for empty is necessary. `is_applicable()` will return right away in this case.

**Comment by liuyuhong66** [comment]
@tchaikov ,tks, just like you said !auth_url.empty() is unnecessary here

**Comment by cbodley** [comment]
passed rgw suite: http://pulpito.ceph.com/cbodley-2017-05-18_16:35:11-rgw-wip-cbodley-testing---basic-mira

**Pull request merged**